Dockerfile

```
@@ -27,8 +27,13 @@ FROM --platform=$BUILDPLATFORM ${GOLANG_IMAGE} AS base
 COPY --from=xx / /
 RUN echo 'Binary::apt::APT::Keep-Downloaded-Packages "true";' > /etc/apt/apt.conf.d/keep-cache
 ARG APT_MIRROR
-RUN sed -ri "s/(httpredir|deb).debian.org/${APT_MIRROR:-deb.debian.org}/g" /etc/apt/sources.list \
- && sed -ri "s/(security).debian.org/${APT_MIRROR:-security.debian.org}/g" /etc/apt/sources.list
+RUN <<EOT
+    set -e
+    if [ -n "${APT_MIRROR}" ]; then
+        sed -ri "s/(httpredir|deb).debian.org/${APT_MIRROR:-deb.debian.org}/g" /etc/apt/sources.list
+        sed -ri "s/(security).debian.org/${APT_MIRROR:-security.debian.org}/g" /etc/apt/sources.list
```

```suggestion
        sed -ri "s/(httpredir|deb).debian.org/${APT_MIRROR}/g" /etc/apt/sources.list
        sed -ri "s/(security).debian.org/${APT_MIRROR}/g" /etc/apt/sources.list
```

These defaults don't make sense anymore as the `if` guard will prevent them from having any effect.

---

Also, this might be better as a one-liner at this point, e.g. `test -n "$VAR" && sed`

---

The latest changes here still don't address the fact that the default substitution will never take effect as `-n "$APT_MIRROR"` guarantees that `$APT_MIRROR` will be set and `${APT_MIRROR:-default}` will never use the `default` value. I think my suggested version with a single `sed` should be preferred.

---

Ah I misread your review sorry :see_no_evil: